bloom-housing/bloom
# feat: auto publish lotteries

This PR addresses [#(4201)](https://app.zenhub.com/workspaces/bloom-5dc32d7144bd400001315dac/issues/gh/bloom-housing/bloom/4201)


## Description

For an opted in lottery listing which has been released to partners, when the lottery start date comes to pass, the system should autopublish the results to the public if partners haven't done so by end of day. This will be done at 11:59PM PST. On publishing the result, emails to associated partners, admin and applicants should be sent and an activity log should be created. Should appear in the history log as `by System`

Additional changes:
- moved out lotteryStatus db update call into helper function
- moved publishing lottery logic into helper function
- added activity log logic to expireLotteries and closeListings cron jobs
- set cron job timezone to PST

## How Can This Be Tested/Reviewed?

Go to or create a listing that is opted in to the system lottery. Set the lottery start date to today. As an admin, close the listing, run the lottery and release it to partners. Using the api, call `lottery/autoPublishResults` endpoint. Refresh listing lottery page and confirm publish appears in the history log correctly.

## Author Checklist:


## Review Process:

- Read and understand the issue
- Ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Either (1) explicitly ask a clarifying question, (2) request changes, or (3) approve the PR, even if there are very small remaining changes, if you don't need to re-review after the updates
